### DISCUSSION TITLE
Package API: Getting HTTP response 404 on unauthorized access

### DISCUSSION THREAD

**Discussion by ghost** [community/community#30898] [Actions] [answered]
When I try to delete a package version via API, which I am not authorized to do, I get a HTTP response 404 (Resource not found) instead of a 403 (Forbidden).

When I use the API on a package version, which I have access to, it works fine. So my use of the API should be correct.

Example: I try to delete a package version from the github organization:

```
$ curl -s -w "\nStatus: %{http_code}\n" -X DELETE \
> -H "Accept: application/vnd.github+json" -H "Authorization: token $TOKEN" \
> https://api.github.com/orgs/github/packages/container/hello-docker/versions/298486
{
  "message": "Package not found.",
  "documentation_url": "https://docs.github.com/rest/reference/packages#delete-a-package-version-for-an-organization"
}

Status: 404
```

The same error happens, when trying to delete package versions from other users, so this is not specific to the github organization.

To prove, that the package version really exists, I only replaced DELETE by GET. This is successful:

```
$ curl -s -w "\nStatus: %{http_code}\n" -X GET \
> -H "Accept: application/vnd.github+json" -H "Authorization: token $TOKEN" \
> https://api.github.com/orgs/github/packages/container/hello-docker/versions/298486
{
  "id": 298486,
  "name": "sha256:c2f1bcc5dbbe68370a9433989c797791878f895980b9cdac0c6b04ab1724ddcd",
  "url": "https://api.github.com/orgs/github/packages/container/hello-docker/versions/298486",
  "package_html_url": "https://github.com/orgs/github/packages/container/package/hello-docker",
  "license": "MIT",
  "created_at": "2020-11-12T22:46:58Z",
  "updated_at": "2020-11-12T22:46:58Z",
  "description": "A simple Container for showcasing the release workflow",
  "html_url": "https://github.com/orgs/github/packages/container/hello-docker/298486",
  "metadata": {
    "package_type": "container",
    "container": {
      "tags": [
        "sha-e2746ece",
        "edge"
      ]
    }
  }
}

Status: 200
```

**Comment by airtower-luna** [comment, marked as answer]
Returning 404 instead of 403 on unauthorized accesses is a common practice when sites want to hide from unauthorized users which resources they don't have access to exist. For example, if you try to access a private repository you don't have access to you get 404, just as if the repository didn't exist. If you got 403 you could check if someone *has* a repository of a specific name, even though you couldn't see its content.

I assume that's what's going on here, even if it doesn't make much sense in the particular case. :sweat_smile:

**Reply by ghost** [reply]
> Returning 404 instead of 403 on unauthorized accesses is a common practice when sites want to hide from unauthorized users which resources they don't have access to exist.

Thanks, that makes sense. I will keep that in mind when I have a look at the status code.
